src/lib/client.ts

```diff
@@ -65,8 +65,19 @@ export class HTTPClient implements IRequestClient {
 
         axiosInstance.interceptors.response.use(
             (response) => response,
-            // eslint-disable-next-line @typescript-eslint/promise-function-async
-            (error) => Promise.reject(error.response.data),
+            // eslint-disable-next-line @typescript-eslint/promise-function-async,@typescript-eslint/tslint/config
+            (error) => {
+                if (!!error.response) {
+                    return Promise.reject(error.response.data);
+                } else if (!!error.request) {
+                    return Promise.reject(error.request);
+                } else {
+                    return Promise.reject({
+                        message: error.message,
+                        code: error.code,
+                    });
```

Comment:
From my testing here, it seems that the last part (the `else` case) is never reached, since the request has been made, and we receive an adjusted version of the response. This means that, in this scenario, there will always be a request. Therefore, if the response is undefined (e.g. DNS error), the error will still contain the request made. This shows that the last part will never be reached.

I am thinking of finding a way to merge the last two cases. This will also clarify the reason of the promise when the request is returned (check [this comment](https://github.com/1Password/connect-sdk-js/pull/16/files#r633460276)), and will also make the code a bit cleaner.

Unless there may be a scenario that may reach the `else` case that I could not think of, I am open to hear suggestions.